### PR TITLE
#2230 Cloud IdP integration

### PR DESCRIPTION
**Pull request by micheleRP**
fixes https://github.com/redpanda-data/documentation-private/issues/2230

This adds high-level info about configuring SSO for Redpanda Cloud:

- Cloud Authentication: [preview](https://deploy-preview-324--redpanda-docs-preview.netlify.app/current/deploy/deployment-option/cloud/security/cloud-authentication/) (adds section SSO and removes bullet that Cloud did not support OIDC)
- What's New in Cloud: [preview](https://deploy-preview-324--redpanda-docs-preview.netlify.app/current/deploy/deployment-option/cloud/whats-new-cloud/)

### CODE REVIEW

**Comment by hcoyote** [comment]
adding for posterity

I don't see the example work flow, so I'm not sure if we tell people what the callback url is that must be used when setting up the IdP side of the configuration. (With the alpha docs, this is what we told people to use when setting up the callback as part of IdP-side application setup)

>  The Redpanda callback location will be:
> 	https://auth.prd.cloud.redpanda.com/login/callback
> 

If that's something we only show in the setup workflow in the UI, we need to document that as a precursor step (or at least a piece of information the IdP admin needs to know when setting up the application connection for redpanda to use, since the IdP admin is probably not the person doing the Redpanda configuration)

**Review by kbatuigas — APPROVED** [review]

**Comment by deniscoady** [comment]
> Configure Authentication: [preview](https://deploy-preview-324--redpanda-docs-preview.netlify.app/current/manage/security/authentication/#oidc) (removes note that Cloud did not support OIDC)

This page seems to focus on "data plane" (or cluster) authentication. Until we can have self-service cluster configurations it's technically _still_ not available for Redpanda Cloud customers. SSO is focusing on the control plane access (organization to Console, but not Redpanda itself).

**Comment by Feediver1** [comment]
@deniscoady Shouldn't this be under Authentication in the Security section? Seems odd that it would go in the RP Cloud Overview. It includes some config info too--again, not overview content. Please advise.

File: modules/deploy/pages/deployment-option/cloud/security/cloud-authentication.adoc [hunk 1]
```diff
@@ -1,16 +1,43 @@
...
+
+=== Email
+
+Redpanda Cloud authenticates data plane users with emails and passwords.
 Passwords are hashed (a one-way function that makes the original value unrecoverable,
 and effectively encrypted) and salted at rest using https://en.wikipedia.org/wiki/Bcrypt[bcrypt^].
 
+=== Single sign-on
+
+You can configure single sign-on (SSO) with an OIDC-based identity provider (IdP) for control plane authentication to Dedicated or BYOC clusters. Redpanda integrates with any OIDC-compliant IdP that supports discovery, including Okta, Auth0, Microsoft Entra, and Active Directory Federation Services (AD-FS). After SSO is enabled for an organization, all new users in the organization are authenticated using SSO.
```
Comment (Feediver1) [inline modules/deploy/pages/deployment-option/cloud/security/cloud-authentication.adoc]: Link "control plane" to the glossary entry for it (for newbies)?

File: modules/deploy/pages/deployment-option/cloud/security/cloud-authentication.adoc [hunk 2]
```diff
@@ -1,16 +1,43 @@
...
 Passwords are hashed (a one-way function that makes the original value unrecoverable,
 and effectively encrypted) and salted at rest using https://en.wikipedia.org/wiki/Bcrypt[bcrypt^].
 
+=== Single sign-on
+
+You can configure single sign-on (SSO) with an OIDC-based identity provider (IdP) for control plane authentication to Dedicated or BYOC clusters. Redpanda integrates with any OIDC-compliant IdP that supports discovery, including Okta, Auth0, Microsoft Entra, and Active Directory Federation Services (AD-FS). After SSO is enabled for an organization, all new users in the organization are authenticated using SSO. 
+
+==== Integrate IdP
+
+On the *Users* page, you can add connections for up to two different IdPs. Enter the client ID, client secret, and discovery URI for the IdP. (See your IdP documentation for these values. The discovery URI may be called the well known URL or the `issuer_url`.) By default, the connection is added in a disabled state. Edit the connection to enable it, and optionally assign a read, write, or admin role to users who log in with the IdP.
```
Comment (Feediver1) [inline modules/deploy/pages/deployment-option/cloud/security/cloud-authentication.adoc]: You must integrate your IdP with Redpanda Cloud to use SSO.
^^Add as first sentence?

**Review by Feediver1 — APPROVED** [review]

**Comment by deniscoady** [comment]
> @deniscoady Shouldn't this be under Authentication in the Security section? Seems odd that it would go in the RP Cloud Overview. It includes some config info too--again, not overview content. Please advise.

How the docs are structured right now the Authentication section is focused on the data plane (cluster) authentication. This feature is exclusively for the control plane. I'm not sure what the best way to structure it, but it would be more aligned with Cloud-only docs that the Redpanda product itself.

File: modules/deploy/pages/deployment-option/cloud/security/cloud-authentication.adoc [hunk 3]
```diff
@@ -1,15 +1,42 @@
...
-and effectively encrypted) and salted at rest using https://en.wikipedia.org/wiki/Bcrypt[bcrypt^].
+Redpanda provides user authentication to your Redpanda organization through email or single sign-on. 
+
+=== Email
+
+Redpanda Cloud can authenticate users with emails and passwords. Passwords are hashed (a one-way function that makes the original value unrecoverable, and effectively encrypted) and salted at rest using https://en.wikipedia.org/wiki/Bcrypt[bcrypt^].
+
+=== Single sign-on
+
+Redpanda Cloud can authenticate users with single sign-on (SSO) to an OIDC-based identity provider (IdP). Redpanda integrates with any OIDC-compliant IdP that supports discovery, including Okta, Auth0, Microsoft Entra, and Active Directory Federation Services (AD-FS). After SSO is enabled for an organization, all new users in the organization are authenticated using SSO.
```
Comment (Mateoc) [inline modules/deploy/pages/deployment-option/cloud/security/cloud-authentication.adoc]: Clarification, they all **can** be authenticated using sso but is up to the user

File: modules/deploy/pages/deployment-option/cloud/security/cloud-authentication.adoc [hunk 4]
```diff
@@ -1,15 +1,42 @@
...
+
+=== Single sign-on
+
+Redpanda Cloud can authenticate users with single sign-on (SSO) to an OIDC-based identity provider (IdP). Redpanda integrates with any OIDC-compliant IdP that supports discovery, including Okta, Auth0, Microsoft Entra, and Active Directory Federation Services (AD-FS). After SSO is enabled for an organization, all new users in the organization are authenticated using SSO. 
+
+==== Integrate IdP
+
+You must integrate your IdP with Redpanda Cloud to use SSO. On the *Users* page, you can add connections for up to two different IdPs. Enter the client ID, client secret, and discovery URI for the IdP. (See your IdP documentation for these values. The discovery URI may be called something different, like the well known URL or the `issuer_url`.) 
+
+By default, the connection is added in a disabled state. Edit the connection to enable it. You can also choose to enable auto-enroll in the connection, which provides any new user signing in from that IdP access to your Redpanda organization. This eliminates the need to manually invite those new users. When you enable auto-enroll, you select to assign a read, write, or admin role to users who log in with that IdP.
```
Comment (Mateoc) [inline modules/deploy/pages/deployment-option/cloud/security/cloud-authentication.adoc]: Note: You have to grant them access from the idp is not automatic, for example in okta you have to assign users/groups in the okta application
@deniscoady not sure if this is worth explaining

**Review by Mateoc — APPROVED** [review]
LGTM
left a couple of notes

File: modules/deploy/pages/deployment-option/cloud/security/cloud-authentication.adoc [hunk 5]
```diff
@@ -19,7 +19,7 @@ Redpanda Cloud can authenticate users with single sign-on (SSO) to an OIDC-based
 
 ==== Integrate IdP
 
-You must integrate your IdP with Redpanda Cloud to use SSO. On the *Users* page, you can add connections for up to two different IdPs. Enter the client ID, client secret, and discovery URI for the IdP. (See your IdP documentation for these values. The discovery URI may be called something different, like the well known URL or the `issuer_url`.)
```
Comment (ghost) [inline modules/deploy/pages/deployment-option/cloud/security/cloud-authentication.adoc]: > you can add connections for up to two different IdPs

It's worth telling customers that if they have a use case which requires more, they should contact us/let us know.

**Review by ghost — APPROVED** [review]

File: modules/deploy/pages/deployment-option/cloud/security/cloud-authentication.adoc (hunk 5 above)
Comment (Feediver1) [reply modules/deploy/pages/deployment-option/cloud/security/cloud-authentication.adoc]: @deniscoady Do you want the docs to prompt users for additional IdP use cases/values? I'm a bit torn, as docs is not typically  where customers make such requests. Please advise.

Comment (deniscoady) [reply modules/deploy/pages/deployment-option/cloud/security/cloud-authentication.adoc]: We plan to call this out in the product so we probably shouldn't make a call to action in the docs.

File: modules/deploy/pages/deployment-option/cloud/security/cloud-authentication.adoc (hunk 4 above)
Comment (deniscoady) [reply modules/deploy/pages/deployment-option/cloud/security/cloud-authentication.adoc]: That is an Okta-specific behavior (you _can_ assign the entire Okta org for example). Anything that is vendor-specific is out of scope for our docs right now, there is just too many vendors to do them all appropriate justice.

Comment (micheleRP) [reply modules/deploy/pages/deployment-option/cloud/security/cloud-authentication.adoc]: > Note: You have to grant them access from the idp is not automatic, for example in okta you have to assign users/groups in the okta application @deniscoady not sure if this is worth explaining

Edited this to: "By default, the connection is added in a disabled state. Edit the connection to enable it. You can also choose to enable auto-enroll in the connection, which provides new users signing in from that IdP access to your Redpanda organization. When you enable auto-enroll, you select to assign a read, write, or admin role to users who log in with that IdP."

**Review by deniscoady — APPROVED** [review]
LGTM!

**Pull request merged**